src/marionette/types.cr

```diff
@@ -22,8 +22,13 @@ module Marionette
     include JSON::Serializable
   end
 
-  record Rect, height : Float64, width : Float64, x : Float64, y : Float64 do
+  struct Rect
```

Comment:
Without this, the following error occurs:
```
In src/marionette/window.cr:105:14

 105 | rect.x = position.x
```

Records are immutable, so I made it a struct. 

Another option would be to use `rect = rect.copy_with x: position.x`, but I thought this was the better option.

Reply:
You could just add setters to the record (which really just expands to a struct definition). 
I'm not sure if mutable instances are actually appropriate for this use case, though.
`copy_with` isn't a bad option anyways.

Reply:
Yeah I personally prefer it to be immutable, and since it's a record it includes `copy_with` out of the box. Just keep it as a record if you could and I'll get this merged.